cylc/flow/platforms.py

```diff
@@ -265,6 +265,12 @@ def platform_from_name(
             platform_data['name'] = platform_name
             return platform_data
 
+    # If platform name in run mode and not otherwise defined:
+    if platform_name == 'SIMULATION':
+        platform_data = deepcopy(platforms['localhost'])
+        platform_data['name'] = 'localhost'
+        return platform_data
+
```

Comment:
Yes and no, the platform wasn't being written to the DB before, now it is, as a consequence this check is now required.

Ronnie hasn't dotted the t's and crossed the i's as this will change with Tim's skip-mode PR.